File: python_translation/vector2.py
"""
Vector2 Python Translation

Original Java file: sources/c13/nim5/ez8/h5_proto/Vector2.java
Package: c13.nim5.ez8.h5_proto

This module provides a 2D vector class with serialization support.
"""

from dataclasses import dataclass, field
from typing import Optional, Dict, Any


@dataclass
class Vector2:
    """
    2D Vector class representing x and y coordinates.

    Attributes:
        x: X coordinate (Long/int)
        y: Y coordinate (Long/int)
    """
    x: Optional[int] = 0
    y: Optional[int] = 0

    def __post_init__(self):
        """Initialize default values."""
        if self.x is None:
            self.x = 0
        if self.y is None:
            self.y = 0

    @staticmethod
    def from_json_object(json_obj: Dict[str, Any]) -> 'Vector2':
        """
        Create a Vector2 instance from a JSON object (dict).

        Args:
            json_obj: Dictionary containing 'x' and 'y' keys

        Returns:
            Vector2 instance

        Raises:
            ValueError: If JSON parsing fails
        """
        vector2 = Vector2()

        # Note: Original Java code uses obfuscated string keys
        # The decrypted keys are likely "x" and "y"
        if 'x' in json_obj and json_obj['x'] is not None:
            vector2.x = int(json_obj['x'])

        if 'y' in json_obj and json_obj['y'] is not None:
            vector2.y = int(json_obj['y'])

        return vector2

    def get_x(self) -> Optional[int]:
        """Get X coordinate."""
        return self.x

    def set_x(self, x: Optional[int]) -> None:
        """Set X coordinate."""
        self.x = x

    def get_y(self) -> Optional[int]:
        """Get Y coordinate."""
        return self.y

    def set_y(self, y: Optional[int]) -> None:
        """Set Y coordinate."""
        self.y = y

    def __str__(self) -> str:
        """
        String representation of the Vector2.

        Returns:
            String in format "Vector2{x=..., y=...}"
        """
        return f"Vector2{{x={self.x}, y={self.y}}}"

    def to_json_object(self) -> Dict[str, Any]:
        """
        Convert Vector2 to a JSON object (dict).

        Returns:
            Dictionary representation of the vector
        """
        return {
            'x': self.x,
            'y': self.y
        }


# Example usage and test
if __name__ == "__main__":
    # Test creation
    v1 = Vector2()
    print(f"Default vector: {v1}")

    # Test with values
    v2 = Vector2(x=10, y=20)
    print(f"Vector with values: {v2}")

    # Test JSON serialization
    json_data = v2.to_json_object()
    print(f"JSON representation: {json_data}")

    # Test JSON deserialization
    v3 = Vector2.from_json_object(json_data)
    print(f"Vector from JSON: {v3}")

    # Test getters/setters
    v3.set_x(100)
    v3.set_y(200)
    print(f"After setters: {v3}")
    print(f"X: {v3.get_x()}, Y: {v3.get_y()}")

File: sources/c13/nim5/ez8/h5_proto/Vector2.java
package c13.nim5.ez8.h5_proto;

import IlIlllIIlI1.IllIIlIIII1;
import IllIIlIIII1.llllIIIIll1;
import org.json.JSONException;
import org.json.JSONObject;

/* loaded from: classes.jar:c13/nim5/ez8/h5_proto/Vector2.class */
public class Vector2 {
    private Long x;
    private Long y;

    public Vector2() {
        this.x = 0L;
        this.y = 0L;
        this.x = 0L;
        this.y = 0L;
    }

    public static Vector2 fromJSONObject(JSONObject jSONObject) throws JSONException {
        Vector2 vector2 = new Vector2();
        byte[] bArr = {-52};
        byte[] bArr2 = {-76, -70, 18, -87, 122, -118, -35, 86};
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        if (jSONObject.has(lllliiiill1.llllIIIIll1(bArr, bArr2)) && !jSONObject.isNull(lllliiiill1.llllIIIIll1(new byte[]{40}, new byte[]{80, 112, -70, 106, -66, -41, 22, 120}))) {
            vector2.x = Long.valueOf(jSONObject.getLong(lllliiiill1.llllIIIIll1(new byte[]{-75}, new byte[]{-51, 47, 75, -77, 39, 105, -119, 63})));
        }
        if (jSONObject.has(lllliiiill1.llllIIIIll1(new byte[]{-112}, new byte[]{-23, 91, -105, -19, 40, -31, -99, -57})) && !jSONObject.isNull(lllliiiill1.llllIIIIll1(new byte[]{-31}, new byte[]{-104, 34, 114, -26, 119, -85, -72, 121}))) {
            vector2.y = Long.valueOf(jSONObject.getLong(lllliiiill1.llllIIIIll1(new byte[]{98}, new byte[]{27, 98, 72, -68, Byte.MAX_VALUE, 28, -27, -87})));
        }
        return vector2;
    }

    public Long getX() {
        return this.x;
    }

    public void setX(Long l) {
        this.x = l;
    }

    public Long getY() {
        return this.y;
    }

    public void setY(Long l) {
        this.y = l;
    }

    public String toString() {
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        return lllliiiill1.llllIIIIll1(new byte[]{126, -119, 121, -124, 52, -108, -3, -87, 80, -47}, new byte[]{40, -20, 26, -16, 91, -26, -49, -46}) + this.x + lllliiiill1.llllIIIIll1(new byte[]{-83, -4, -17, Byte.MIN_VALUE}, new byte[]{-127, -36, -106, -67, -103, -63, Byte.MAX_VALUE, -38}) + this.y + lllliiiill1.llllIIIIll1(new byte[]{-91}, new byte[]{-40, -41, 81, -48, -18, 109, -56, 82});
    }

    public JSONObject toJSONObject() throws JSONException {
        JSONObject jSONObject = new JSONObject();
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{95}, new byte[]{39, -96, -46, -113, 43, -118, -10, -41}), this.x);
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{37}, new byte[]{92, 27, 81, -50, 104, -53, 4, -7}), this.y);
        return jSONObject;
    }
}
